src/java.base/share/classes/java/util/Locale.java

```diff
@@ -398,76 +464,49 @@
- * name of the country suitable for displaying to the user. Similarly,
- * you can use {@code getDisplayLanguage} to get the name of
- * the language suitable for displaying to the user. Interestingly,
- * the {@code getDisplayXXX} methods are themselves locale-sensitive
- * and have two versions: one that uses the default
- * {@link Locale.Category#DISPLAY DISPLAY} locale and one
- * that uses the locale specified as an argument.
+ * <p>During serialization, writeObject writes all fields to the output
+ * stream, including extensions.
  *
- * <p>The Java Platform provides a number of classes that perform locale-sensitive
- * operations. For example, the {@code NumberFormat} class formats
- * numbers, currency, and percentages in a locale-sensitive manner. Classes
- * such as {@code NumberFormat} have several convenience methods
- * for creating a default object of that type. For example, the
- * {@code NumberFormat} class provides these three convenience methods
- * for creating a default {@code NumberFormat} object:
- * {@snippet lang=java :
- *     NumberFormat.getInstance();
- *     NumberFormat.getCurrencyInstance();
- *     NumberFormat.getPercentInstance();
- * }
- * Each of these methods has two variants; one with an explicit locale
- * and one without; the latter uses the default
- * {@link Locale.Category#FORMAT FORMAT} locale:
- * {@snippet lang=java :
- *     NumberFormat.getInstance(myLocale);
- *     NumberFormat.getCurrencyInstance(myLocale);
- *     NumberFormat.getPercentInstance(myLocale);
- * }
- * A {@code Locale} is the mechanism for identifying the kind of object
- * ({@code NumberFormat}) that you would like to get. The locale is
- * <STRONG>just</STRONG> a mechanism for identifying objects,
- * <STRONG>not</STRONG> a container for the objects themselves.
- *
- * <h3>Compatibility</h3>
- *
- * <p>In order to maintain compatibility, Locale's
- * constructors retain their behavior prior to the Java Runtime
- * Environment version 1.7.  The same is largely true for the
- * {@code toString} method. Thus Locale objects can continue to
- * be used as they were. In particular, clients who parse the output
- * of toString into language, country, and variant fields can continue
- * to do so (although this is strongly discouraged), although the
- * variant field will have additional information in it if script or
- * extensions are present.
+ * <p>During deserialization, readResolve adds extensions as described
+ * in {@linkplain ##special_cases_constructor Special Cases}, only
+ * for the two cases th_TH_TH and ja_JP_JP.
+ *
+ * <h2>Compatibility</h2>
+ * @implNote
+ * <p>Applications on the latest release of the reference implementation using the recommended
+ * {@code Locale} API may not be concerned with the following compatibility commentary.
+ * <h3><a id="locale_behavior">Locale Behavior</a></h3>
+ * In order to maintain compatibility, Locale's
+ * (deprecated) constructors retain their behavior prior to the Java Runtime
```

Comment:
I think the factory methods `of()` should also be mentioned here (in the same category with the constructors)